### PR TITLE
Remove thunder.compile and tests

### PR DESCRIPTION
**Pull request by t-vi**
based on the awesome #837 . As a PR to get a picture of what the CI makes of it.

### CODE REVIEW

File: thunder/tests/test_grad.py [hunk 1]
```diff
@@ -239,8 +239,16 @@ def check_jvp(f, *primals, comp, executor):
...
-    actual_p, actual_t = executor.make_callable_legacy(jvp(f))(primals, tangents)
-    expected_p, expected_t = numerical_jvp(executor.make_callable_legacy(f))(primals, tangents)
+    initial_trace_jvp_f = thunder.trace()(jvp(f), primals, tangents)
+    actual_p, actual_t = executor.make_callable(initial_trace_jvp_f.python_callable(), disable_torch_autograd=True)(
+        primals, tangents
+    )
+
+    # dirty little trick for speed: skip the prologue
+    jf = executor.make_callable(f, disable_torch_autograd=True)
+    comp_f = thunder.compile_data(jf).get_computation_and_inputs(*primals)[0].computation_fn
```
Comment (t-vi) [inline thunder/tests/test_grad.py:249]: skipping the prologue here (and reducing the number of cases for polygamma) made things on the windows runner go from >= 35 min to ~28:30 min.

File: thunder/tests/test_grad.py [hunk 2]
```diff
@@ -313,12 +321,18 @@ def check_vjp(f, *primals, comp, executor="torch"):
     make = partial(make_tensor_like, low=0, high=1)
 
     u = tree_map(make, primals)
-    outs_p, J_u = numerical_jvp(executor.make_callable_legacy(f, disable_torch_autograd_support=True))(primals, u)
+
+    # dirty little trick for speed: skip the prologue
```
Comment (t-vi) [inline thunder/tests/test_grad.py:325]: same trick

File: .github/workflows/ci-testing.yml [hunk 3]
```diff
@@ -93,7 +93,7 @@ jobs:
               --ignore=thunder/tests/distributed \
               -v --datefmt="%Y%m%d-%H:%M:%S.%f" \
               --random-order-seed=$GITHUB_RUN_ID \
-              -n 2 --durations=250
+              -n 4 --durations=250
```
Comment (t-vi) [inline .github/workflows/ci-testing.yml:96]: so the runners for linux and win have both 4 CPUs, increasing this to 4 make Windows go from ~28 to ~22 and Linux is now slowest at ~23:30 (before the trick for jvp).

**Comment by t-vi** [comment]
@IvanYashchuk @kiya00 this has your branch with main merged and the additions pointed out in the code comments.
How do you want to move this forward?
- you just pick the bits that you want,
- we merge this into your branch,
- we merge this into main.

**Review by IvanYashchuk — APPROVED** [review]
Thank you! Let's merge this PR.

**Pull request merged**